autopush/endpoint.py

```diff
@@ -867,3 +868,67 @@ def _load_params(self):
             return json.loads(self.request.body)
         except ValueError:
             return {}
+
+
+class BridgeMessageHandler(RegistrationHandler):
+    cors_methods = "PUT"
+
+    @cyclone.web.asynchronous
+    def put(self, uaid):
+        """HTTP PUT
+
+        Process message states for a UAID
+
+        """
+        self.start_time = time.time()
+        if not self._validate_auth(uaid):
+            return self._write_unauthorized_response(
+                message="Invalid Authentication")
+        params = self._load_params()
+        if 'nak' in params:
+            d = deferToThread(self._record_naks, uaid, naks=params['nak'])
+        else:
+            d = Deferred()
+            d.callback(True)
+        d.addCallback(self._success)
+        d.addErrback(self._response_err)
+        return d
+
+    def _record_naks(self, o_uaid, naks):
+        """ Record the NAKs from the client
+
+        """
+        for nak in naks:
+            try:
+                result = self.ap_settings.fernet.decrypt(
+                    nak['id'].encode('utf8'))
+                kind, uaid, chid = result.split(":")
+                if kind != 'm':
+                    self.log.error(
```

Comment:
So if we see this logged, then it means our app is malfunctioning, and we'll need to open a bug to fix something that caused it, right?

I'm just trying to make sure we're logging actionable things, rather than adding more noise to our logstream (which is annoying noisy already).